proto/expr.proto

```diff
@@ -472,17 +472,21 @@ message UserDefinedFunction {
   string language = 4;
   // For external UDF: the link to the external function service.
   // For WASM UDF: the link to the wasm binary file.
-  string link = 5;
+  optional string link = 5;
   // An unique identifier for the function.
   // For external UDF, it's the name of the function in the external function service.
   // For WASM UDF, it's the name of the function in the wasm binary file.
-  string identifier = 6;
+  // For JavaScript UDF, it's the name of the function.
+  optional string identifier = 6;
+  // For JavaScript UDF, it's the body of the function.
+  optional string body = 7;
 }
 
 // Additional information for user defined table functions.
 message UserDefinedTableFunction {
   repeated data.DataType arg_types = 3;
   string language = 4;
-  string link = 5;
-  string identifier = 6;
+  optional string link = 5;
+  optional string identifier = 6;
+  optional string body = 7;
```

Comment:
Does `optinal` means we allow user input the UDF via link or identifier or body?

Reply:
Previously for external Python and Java UDF, `link` and `identifier` are required. But for the new SQL and JavaScript UDF, the code is embedded as `body`. So now we check the existence of these fields according to the `language`.

Reply:
Is `string` field implicitly `optional` before? If not, doesn't this cause imcompatibility?

Reply:
IIUC, without the `optional` it's still "optional" (with the semantics of default==null), and `optional` means "explicit presence". 

Changing it is binary-safe. But may or may not be safe according to the app logic.. https://protobuf.dev/programming-guides/field_presence/#considerations-for-change-compatibility

Reply:
Yes, adding an `optional` only changes the Rust code generated by prost. I think it is backward-compatible but not forward-compatible.